packages/gtmapi/lmsrvcore/api/mutations/chunkupload.py

```diff
@@ -92,13 +75,38 @@ def validate_args(args):
 
     @staticmethod
     def py_secure_filename(filename: str) -> str:
-        # Take any non-alphanumeric (and some limited special chars) and replace with _
-        # Also note, max filename size is 4096 chars
-        s = lambda n: n if (n.isalnum() or n in '._-+=') else '_'
-        safe_fname = ''.join([s(c) for c in os.path.basename(filename)])[:255]
-        if safe_fname != filename:
-            logger.info(f"Renaming unsafe filename `{filename}` to `{safe_fname}`")
-        return safe_fname
+        """Method to clean up provided filenames to be safe, relative paths
+
+        This function removes leading slashes, control characters, '..' and '.' in directories, and replaces \/:*"<>|?
+        with underscores.
+
+
+        Args:
+            filename: Filename to sanitize
+
+        Returns:
+            str
+        """
+        # Completely remove control characters
+        safe_filename = "".join(c for c in filename if unicodedata.category(c)[0] != "C")
+
+        # Remove leading slash if attempting an absolute path
+        if safe_filename[0] == os.path.sep:
```

Comment:
This is problematic on Windows 10, which accepts \ and /. If we're not supporting windows 10, I think it makes more sense to just use:
```suggestion
        if safe_filename[0] in r'\/':
```
(note that order matters there because of the special case of escaping the quote)

Reply:
this code is running in the container, so os.path.sep is always `/`...but I guess fine to be explicit